docs/servicedns-with-externaldns.md

```diff
@@ -0,0 +1,221 @@
+        matchLabels:
+          app: nginx
+      template:
+        metadata:
+          labels:
+            app: nginx
+        spec:
+          containers:
+          - image: nginx
+            name: nginx
+---
+apiVersion: core.federation.k8s.io/v1alpha1
+kind: FederatedDeploymentPlacement
+metadata:
+  name: test-deployment
+spec:
+  clusterNames:
+  - cluster2
+  - cluster1
+---
+apiVersion: core.federation.k8s.io/v1alpha1
+kind: FederatedService
+metadata:
+  name: test-service
+spec:
+  template:
+    spec:
+      selector:
+        app: nginx
+      type: LoadBalancer
+      ports:
+        - name: http
+          port: 80
+---
+apiVersion: core.federation.k8s.io/v1alpha1
+kind: FederatedServicePlacement
+metadata:
+  name: test-service
+spec:
+  clusterNames:
+  - cluster2
+  - cluster1
+EOF
+```
+
+Check the status of all the resources in each cluster by running:
+```bash
+for r in deployment service; do
+    for c in cluster1 cluster2; do
+        echo; echo ------------ ${c} resource: ${r} ------------; echo
+        kubectl --context=${c} get ${r}
+        echo; echo
+    done
+done
+```
+
+It may take a few minutes for the `EXTERNAL-IP` field of each `Service` to be populated. Next, create the
+`MultiClusterServiceDNSRecord`:
+```bash
+cat <<EOF | kubectl create -f -
+apiVersion: multiclusterdns.federation.k8s.io/v1alpha1
+kind: MultiClusterServiceDNSRecord
+metadata:
+  name: test-service
+spec:
+  federationName: federation-system
+  dnsSuffix: external.example.com
+  recordTTL: 300
+EOF
+```
+
+The DNS Endpoint controller will use the address from each `Service` to populate the `targets` field of the
+`DNSEndpoint` object. For example:
+ ```bash
+$ kubectl get dnsendpoint -o yaml
+apiVersion: multiclusterdns.federation.k8s.io/v1alpha1
+kind: DNSEndpoint
+metadata:
+  creationTimestamp: 2018-10-12T21:41:12Z
+  generation: 1
+  name: service-test-service
+  namespace: default
+  resourceVersion: "755496"
+  selfLink: /apis/multiclusterdns.federation.k8s.io/v1alpha1/namespaces/default/dnsendpoints/service-test-service
+  uid: 89c18705-ce67-11e8-bebb-42010a8a00b8
+spec:
+  endpoints:
+  - dnsName: test-service.default.federation-system.svc.external.example.com
+    recordTTL: 300
+    recordType: A
+    targets:
+    - $CLUSTER1_SERVICE_IP
+    - $CLUSTER2_SERVICE_IP
+  - dnsName: test-service.default.federation-system.svc.us-west1-b.us-west1.external.example.com
+    recordTTL: 300
+    recordType: A
+    targets:
+    - $CLUSTER1_SERVICE_IP
+    - $CLUSTER2_SERVICE_IP
+  - dnsName: test-service.default.federation-system.svc.us-west1.external.example.com
+    recordTTL: 300
+    recordType: A
+    targets:
+    - $CLUSTER1_SERVICE_IP
+    - $CLUSTER2_SERVICE_IP
+status: {}
+```
+
+The ExternalDNS controller is watching for `DNSEndpoint` objects and creates multiple `A` and `TXT` resources records
+for each `FederatedService` in the configured DNS provider.
+```bash
+gcloud dns record-sets list --zone="external-example-com"
```

Comment:
Add a `$` for this command to distinguish the command line and the return content. Ditto for the command of `dig`.

Reply:
ack